cpp/tests/strings/integers_tests.cu

```diff
@@ -164,20 +164,38 @@ TEST_F(StringsConvertTest, HexToInteger)
     h_strings.end(),
     thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; }));
 
-  std::vector<int32_t> h_expected;
-  for (auto itr = h_strings.begin(); itr != h_strings.end(); ++itr) {
-    if (*itr == nullptr)
-      h_expected.push_back(0);
-    else
-      h_expected.push_back((int)std::stol(std::string(*itr), 0, 16));
+  {
+    std::vector<int32_t> h_expected;
+    for (auto itr = h_strings.begin(); itr != h_strings.end(); ++itr) {
+      if (*itr == nullptr)
+        h_expected.push_back(0);
+      else
+        h_expected.push_back((int)std::stol(std::string(*itr), 0, 16));
```

Comment:
C-style casts are doing more than they should (they are all-encompassing). Consider `static_cast<int>()` instead, which should be more efficient.